src/main/java/com/minecolonies/colony/Field.java

```diff
@@ -0,0 +1,541 @@
+    /**
+     * The colony of the field.
+     */
+    private final Colony   colony;
+    /**
+     * The fields location.
+     */
+    private       BlockPos location;
+    /**
+     * Has the field be taken by any worker?
+     */
+    private boolean taken = false;
+
+    /**
+     * Checks if the field needsWork (Hoeig, Seedings, Farming etc).
+     */
+    private boolean needsWork = false;
+
+    /**
+     * Has the field been planted?
+     */
+    private FieldStage fieldStage = FieldStage.EMPTY;
+
+    /**
+     * The length to plus x of the field.
+     */
+    private int lengthPlusX;
+
+    /**
+     * The width to plus z of the seed.
+     */
+    private int widthPlusZ;
+
+    /**
+     * The length to minus xof the field.
+     */
+    private int lengthMinusX;
+
+    /**
+     * The width to minus z of the seed.
+     */
+    private int widthMinusZ;
+
+    /**
+     * The inventorySlot of the field.
+     */
+    private InventoryField inventory;
+
+    /**
+     * Private constructor to create field from NBT.
+     *
+     * @param colony the colony the field belongs to.
+     */
+    private Field(Colony colony)
+    {
+        super();
+        this.colony = colony;
+    }
+
+    /**
+     * Creates an instance of our field container, this may be serve to open the GUI.
+     *
+     * @param inventory       the field inventory.
+     * @param playerInventory the player inventory.
+     * @param world           the world.
+     * @param location        the position of the field.
+     */
+    public Field(InventoryField inventory, InventoryPlayer playerInventory, World world, BlockPos location)
+    {
+        super();
+        this.colony = ColonyManager.getColony(world, location);
+        this.location = location;
+
+        this.inventory = inventory;
+
+        addSlotToContainer(new Slot(inventory, 0, X_OFFSET, Y_OFFSET));
+
+        //Ddd player inventory slots
+        // Note: The slot numbers are within the player inventory and may be the same as the field inventory.
+        int i;
+        for (i = 0; i < PLAYER_INVENTORY_ROWS; i++)
+        {
+            for (int j = 0; j < PLAYER_INVENTORY_COLUMNS; j++)
+            {
+                addSlotToContainer(new Slot(
+                        playerInventory,
+                        j + i * PLAYER_INVENTORY_COLUMNS + PLAYER_INVENTORY_COLUMNS,
+                        PLAYER_INVENTORY_INITIAL_X_OFFSET + j * PLAYER_INVENTORY_OFFSET_EACH,
+                        PLAYER_INVENTORY_INITIAL_Y_OFFSET + i * PLAYER_INVENTORY_OFFSET_EACH
+                ));
+            }
+        }
+
+        for (i = 0; i < PLAYER_INVENTORY_COLUMNS; i++)
+        {
+            addSlotToContainer(new Slot(
+                    playerInventory, i,
+                    PLAYER_INVENTORY_INITIAL_X_OFFSET + i * PLAYER_INVENTORY_OFFSET_EACH,
+                    PLAYER_INVENTORY_HOTBAR_OFFSET
+            ));
+        }
+        calculateSize(world, location.down());
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this call from a constructor to the overridable "calculateSize" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1699)